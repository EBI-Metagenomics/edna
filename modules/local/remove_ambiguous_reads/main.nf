
process REMOVE_AMBIGUOUS_READS {
    // Run DADA2 pipeline including read-tracking
    tag "$meta.id"
    label 'process_medium'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*noambig*fastq.gz"), emit: noambig_out
    path "versions.yml"                       , emit: versions

    script:
    """
    if [[ ${meta.single_end} = true ]]; then
        remove_ambiguous_reads -f $reads -s ${meta.id}
    else 
        remove_ambiguous_reads -f ${reads[0]} -r ${reads[1]} -s ${meta.id}
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: ${params.mpt_version}
    END_VERSIONS
    """
}
