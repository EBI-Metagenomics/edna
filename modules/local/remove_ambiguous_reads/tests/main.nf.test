nextflow_process {

    name "Test Process REMOVE_AMBIGUOUS_READS"
    script "../main.nf"
    process "REMOVE_AMBIGUOUS_READS"
    tag "modules"
    tag "remove_ambiguous_reads"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ 
                        file('${projectDir}/modules/local/remove_ambiguous_reads/tests/data/test_1.cutadapt.fastq.gz', checkIfExists: true),
                        file('${projectDir}/modules/local/remove_ambiguous_reads/tests/data/test_2.cutadapt.fastq.gz', checkIfExists: true) 
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.noambig_out).match("noambig_out") }
            )
        }

    }

}
