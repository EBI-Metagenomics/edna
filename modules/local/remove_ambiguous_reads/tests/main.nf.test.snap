{
    "noambig_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_noambig_1.fastq.gz:md5,1fcc28fbb35ff646fbf61d88dad454f6",
                        "test_noambig_2.fastq.gz:md5,957292d0eee42dbe63f5b11dd9aa7e0f"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T16:38:56.338472452"
    }
}